lib/rgeo/geo_json/collection_methods.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module RGeo
+  # This module is here to fill the gap between what is a GeometryCollection (GIS)
+  # and a FeatureCollection (GeoJSON).
+  #
+  # Note for contributors, you can rely on `@features` to be defined and
+  # you can consider working with an Enumerable wrapping `@features`. See
+  # GeoJSON::FeatureCollection.
+  module GeoJSON::CollectionMethods
+    # There is tight coupling between {FeatureCollection} and this, hence the
+    # guard.
+    private_class_method def self.included(base)
+      return if base.to_s == "RGeo::GeoJSON::FeatureCollection"
+
+      raise Error::RGeoError, "#{self.class} must only be used by FeatureCollection"
+    end
+
+    private def method_missing(symbol, *args)
+      return super unless any? { |feature| feature.respond_to?(symbol) }
+
+      raise Error::UnsupportedOperation, "Method FeatureCollection##{symbol} " \
+        "is not defined. You may consider filing an issue or opening a pull " \
+        "request at https://github.com/rgeo/rgeo-geojson"
+    end
+
+    def contains?(geometry)
+      any? { |feature| feature.contains?(geometry) }
+    end
+
+    def intersects?(geometry)
+      any? { |feature| feature.intersects?(geometry) }
+    end
+  end
```

Comment:
Are most of the geometric predicates going to have this format? Could we just have an array of them and define them all like so?

```rb
methods = [:contains?, :intersects?] # add more
methods.each do |method|
  define_method(method) do |geom|
    any? {|feature| feature.public_send(:method, geometry)}
  end
end
```

Not a big deal if this won't work, but it could be worth it to look into and see if the assumption that this format of checking (i.e. using `any?` on each feature) would validly apply for all spatial predicates.

Reply:
@keithdoggett if that's ok with you I'd rather keep both that way for now, since there are only two. When one will need new predicates we can consider merging.

The issue I see is that we may quickly get overly complicated meta programming due to specification being a bit different. For instance, `any?` gives a boolean, but a lot of geometries operations must return a geometry. Some predicates may need to maybe use `all?` as well.

That would allow quick merging here, and a new PR dedicated to that subject whenever it becomes needed.

Reply:
@BuonOmo yeah that's fine if you want to keep it like this. I agree the analysis methods may be harder to implement since they can have different parameters. I was thinking that this could work for the predicates, but it is easy to find an example that wouldn't work. `disjoint?` would need `all?` while `intersects?` would need `any?`.